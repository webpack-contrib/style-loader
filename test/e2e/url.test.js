/* eslint-disable */
import path from 'path';
import dom from '../helpers/dom';
import webpack from '../helpers/compiler';

describe('E2E', () => {
  test('URL', async () => {
    const config = {
      rules: [
        {
          test: /\.css$/,
          use: [
            { loader: path.resolve(__dirname, '../../src/url') },
            {
              loader: 'file-loader',
              options: {}
            }
          ]
        }
      ]
    };

    const stats = await webpack('fixture.js', config);
    const { assets } = stats.compilation;

    const scripts = {
      main: assets['main.js'].source(),
      runtime: assets['runtime.js'].source()
    };

    const { window } = dom([ scripts.runtime, scripts.main ]);

    const links = window.document.head.innerHTML.trim();

    expect(links).toMatchSnapshot();
  });
});
