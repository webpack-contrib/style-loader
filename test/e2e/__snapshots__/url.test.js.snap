// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E URL 1`] = `"<link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"9d518ff9795764444f4768fe49d8e628.css\\">"`;
